pkg/option/config.go

```diff
@@ -2346,8 +2353,16 @@ func (c *DaemonConfig) Populate() {
 
 	c.populateDevices()
 
-	if nativeCIDR := viper.GetString(IPv4NativeRoutingCIDR); nativeCIDR != "" {
-		c.ipv4NativeRoutingCIDR = cidr.MustParseCIDR(nativeCIDR)
+	if c.IPAM == ipamOption.IPAMOperator {
+		var nativeCIDR string
+		c.IPAMClusterPoolIPv4RoutingCIDR = viper.GetStringSlice(operatorOption.IPAMOperatorV4CIDR)
+		if nativeCIDR = strings.Join(c.IPAMClusterPoolIPv4RoutingCIDR[:0], ""); nativeCIDR != "" {
+			c.ipv4NativeRoutingCIDR = cidr.MustParseCIDR(nativeCIDR)
+		}
+	} else {
```

Comment:
Interesting, I didn't know that `ipv4NativeRoutingCIDR` could only have 1 CIDR. This needs to be addressed first since we allow to have more than one CIDR in the `IPAMClusterPoolIPv4RoutingCIDR`. The problem rises with the fact that `native-routing-cidr` is a flag that can only receive a single CIDR and we can't change its behavior. Perhaps we would need to deprecate the `native-routing-cidr` and introduce a `native-routing-cidrs`. We should discuss this in the weekly meeting.

Reply:
@aanm, Yes, I had this question while I was working on it, since it was a array to a single value match. Ok, let us discuss it in the weekly meeting.

Reply:
I understand that we're looking for @brb 's input on this, he's currently OOO but will be back next week.

Reply:
Ok, thanks will wait for the reply.

Reply:
@soumynathan were you able to get the feedback you need from @brb?

Reply:
We discussed during the meeting, that the current implementation of the BPF masq assumes that there will be an exactly one native routing CIDR. To overcome this limitation, we could reuse the existing BPF ip-masq-agent infrastructure, and to populate the CIDRs into its LPM BPF maps.

Reply:
> We discussed during the meeting, that the current implementation of the BPF masq assumes that there will be an exactly one native routing CIDR. To overcome this limitation, we could reuse the existing BPF ip-masq-agent infrastructure, and to populate the CIDRs into its LPM BPF maps.

@brb @aanm, so do we need to modify this PR or drop it and work on a different one.

Reply:
I suggest to make a final decision during next week's community meeting.

Reply:
@brb Thanks,